app/src/main/java/com/example/healthinspector/Fragments/ProductFinderFragment.java

```diff
@@ -0,0 +1,223 @@
+import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.tasks.OnSuccessListener;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.parceler.Parcels;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+
+public class ProductFinderFragment extends Fragment implements OnMapReadyCallback {
+
+    private FragmentProductFinderBinding binding;
+    public static final String TAG = "ProductFinderFragment";
+    private int count;
+    private ArrayList<JSONObject> locations;
+    private ActivityResultLauncher<String> requestPermissionLauncher;
+    private LocationManager locationManager;
+    private Location lastLocation;
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        binding = FragmentProductFinderBinding.inflate(inflater, container, false);
+        View view = binding.getRoot();
+
+        locations = KrogerLocationCacher.getInstance().getKrogerLocations();
+        SupportMapFragment mapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+
+        Bundle bundle = getArguments();
+        RecommendedProduct recommendedProduct = Parcels.unwrap(bundle.getParcelable(Constants.RECOMMENDED_PRODUCT));
+        ScannedProduct scannedProduct = Parcels.unwrap(bundle.getParcelable(Constants.SCANNED_PRODUCT));
+        count = 0;
+
+        if(locations == null || locations.size() == 0){
+            Toast.makeText(requireContext(), getString(R.string.no_nearby_locations), Toast.LENGTH_LONG).show();
+            //navigate user back to product details fragment and finish this fragment
+            FragmentManager fragmentManager = getActivity().getSupportFragmentManager();
+            FragmentTransaction fragmentTransaction =  fragmentManager.beginTransaction();
+            fragmentTransaction.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+            fragmentManager.popBackStack();
+            fragmentTransaction.commit();
+            return null;
+        }
+        //API request to check all locations for the recommended product's availability
+        for(int i = 0; i < locations.size(); i++){
+            count = i;
+            OkHttpClient client = new OkHttpClient();
+            Request request = null;
+            try {
+                request = new Request.Builder()
+                        .url("https://api.kroger.com/v1/products?filter.brand=" + "Oreo" + "&filter.term=" + "cookie" + "&filter.fulfillment=ais" + "&filter.locationId=" + locations.get(count).getString(Constants.LOCATION_ID))
```

Comment:
Since we are using dummy values here, we should add a TODO comment to fix it later